### PR TITLE
Dedentation package

### LINKED ISSUE #5: Add dedent package
Awesome work with this package, really what I was looking for. I could clean up a lot of duplicated helper functions by replacing it with this library.

I have a feature request: Dedentation. It is quite helpful. I currently use a helper function and spotted some code in https://github.com/cli/cli doing the same. It is essentially:

```go
func dedent(s string) string {
	lines := strings.Split(s, "\n")
	minIndent := -1

	for _, l := range lines {
		if len(l) == 0 {
			continue
		}

		indent := len(l) - len(strings.TrimLeft(l, " "))
		if minIndent == -1 || indent < minIndent {
			minIndent = indent
		}
	}

	if minIndent <= 0 {
		return s
	}

	var buf bytes.Buffer
	for _, l := range lines {
		fmt.Fprintln(&buf, strings.TrimPrefix(l, strings.Repeat(" ", minIndent)))
	}
	return strings.TrimSuffix(buf.String(), "\n")
}
```

The algorithm behind that is simple: Find the longest indentation possible for all lines and dedent each line by that.

If you like the idea, I can make a PR and add some tests.

### PR DESCRIPTION
**Pull request by lukasmalkmus**
This adds a `dedent` package. Closes #5.

Instead of the implementation provided in the issue, I use a `strings.Builder` for Go 1.10 and onwards, which was a very easy performance improvement:

```
name      old time/op    new time/op    delta
Dedent-8     835ns ± 3%     656ns ± 8%  -21.50%  (p=0.000 n=9+10)

name      old alloc/op   new alloc/op   delta
Dedent-8      344B ± 0%      328B ± 0%   -4.65%  (p=0.000 n=10+10)

name      old allocs/op  new allocs/op  delta
Dedent-8      11.0 ± 0%       9.0 ± 0%  -18.18%  (p=0.000 n=10+10)
```

### CODE REVIEW

**Comment by lukasmalkmus** [comment]
Hm, the build failure seems related to coveralls, not my code additions.

**Comment by lukasmalkmus** [comment]
Ups, got it. CI tests against very old go versions. I don't know your stance regarding backwards compatibility but I just made it work by using a `bytes.Buffer` for Go 1.9 and before.

**Comment by muesli** [comment]
@lukasmalkmus I think it's fine to drop Go 1.10 support in the next release. Go ahead, use the StringBuilder and just remove 1.10 from the CI config.

**Comment by lukasmalkmus** [comment]
Alright, done. Ditched Go 1.9 support and lower, updated CI to test 1.14 and 1.15 and updated the only dependency to something more recent.

**Comment by lukasmalkmus** [comment]
I just ran `go mod tidy`. Maybe it is a good idea to squash before merging.

File: dedent/dedent.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package dedent
+
+import (
+	"strings"
+)
+
+// String automatically detects the maximumen identation shared by all lines and
```
Comment (muesli) [inline dedent/dedent.go:7]: Typos: `maximum` and `indentation`.

**Review by muesli — CHANGES_REQUESTED** [review]
Looking good, just the two typos I could spot.

**Comment by lukasmalkmus** [comment]
Awesome, thanks for you quick response. Do you mind cutting a new release, soon?

Cheers.

**Comment by muesli** [comment]
No, thank you for the contribution! I'll release a `v0.2.0` tonight!

**Pull request merged**